### PR TITLE
Adaptive Compression Training Loop support from NNCF source

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
Removed ACTL implementation from DOR and replaced it with NNCF implementation

(temporary NNCF fork commit in requirements)

Ticket: 95048

DOR PR: https://github.com/openvinotoolkit/deep-object-reid/pull/241
NNCF PR: https://github.com/openvinotoolkit/nncf/pull/1401

Experiment results on CIFAR10 dataset with `external/model-preparation-algorithm/configs/classification/mobilenet_v3_small_cls_incr/template_experiment.yaml` config and  `batch_size=512`

| | Before      | After |
| - | ----------- | -- |
| Compression rate | 60.85% | 63.73%|
| Absolute accuracy drop | 0.9% | 0.72% |
| Number of epochs | 200 | 200 |

### CODE REVIEW

**Comment by nikita-savelyevv** [comment]
Closing as DOR is being deprecated
